components/tooltip/Tooltip.js

```diff
@@ -0,0 +1,271 @@
+            top: top + height / 2 + yOffset,
+            left: left + width + xOffset,
+            position,
+          };
+        }
+
+        return undefined;
+      }
+
+      handleMouseEnter = event => {
+        this.activate(this.calculatePosition(event.currentTarget));
+
+        if (this.props.onMouseEnter) {
+          this.props.onMouseEnter(event);
+        }
+      };
+
+      handleMouseLeave = event => {
+        this.deactivate();
+
+        if (this.props.onMouseLeave) {
+          this.props.onMouseLeave(event);
+        }
+      };
+
+      handleClick = event => {
+        if (this.props.tooltipHideOnClick && this.state.active) {
+          this.deactivate();
+        }
+
+        if (this.props.tooltipShowOnClick && !this.state.active) {
+          this.activate(this.calculatePosition(event.currentTarget));
+        }
+
+        if (this.props.onClick) {
+          this.props.onClick(event);
+        }
+      };
+
+      render() {
+        const { active, left, top, position, visible } = this.state;
+        const positionClass = `tooltip${position.charAt(0).toUpperCase() + position.slice(1)}`;
+        const {
+          children,
+          className,
+          onClick, // eslint-disable-line no-unused-vars
+          onMouseEnter, // eslint-disable-line no-unused-vars
+          onMouseLeave, // eslint-disable-line no-unused-vars
+          tooltip,
+          tooltipColor,
+          tooltipDelay, // eslint-disable-line no-unused-vars
+          tooltipHideOnClick, // eslint-disable-line no-unused-vars
+          tooltipIcon,
+          tooltipPosition, // eslint-disable-line no-unused-vars
+          tooltipShowOnClick, // eslint-disable-line no-unused-vars
+          tooltipSize,
+          ...other
+        } = this.props;
+
+        const _className = cx(theme.tooltip, theme[tooltipColor], theme[tooltipSize], {
+          [theme.tooltipActive]: active,
+          [theme[positionClass]]: theme[positionClass],
+        });
+
+        const childProps = {
+          ...other,
+          className,
+          onClick: this.handleClick,
+          onMouseEnter: this.handleMouseEnter,
+          onMouseLeave: this.handleMouseLeave,
+        };
+
+        const shouldPass = typeof ComposedComponent !== 'string' && defaultPassthrough;
+        const finalProps = shouldPass ? { ...childProps } : childProps;
+
+        return React.createElement(
+          ComposedComponent,
+          finalProps,
+          children,
+          visible && (
+            <Portal>
+              <div
+                ref={node => {
+                  this.tooltipNode = node;
+                }}
+                className={_className}
+                data-teamleader-ui="tooltip"
+                style={{ top, left }}
+              >
+                <div className={theme.tooltipInner}>
+                  <div className={theme.tooltipArrow} />
+                  {tooltipIcon && <div className={theme.tooltipIcon}>{tooltipIcon}</div>}
```

Comment:
Why do you have a separate `tooltipIcon` property? An icon can also be inside the tooltip element rendered the line below.

Reply:
We chose to always treat the icons as a separate prop, so we could:
- create consistency over all the components
- apply css to the icon specificly
- do some checks with the prop to render something different